sim/warlock/talents_demonology.go

```diff
@@ -74,22 +93,20 @@ func (warlock *Warlock) registerImpendingDoom() {
 
 	impendingDoomProcChance := 0.05 * float64(warlock.Talents.ImpendingDoom)
 
-	warlock.NightfallProcAura = warlock.RegisterAura(core.Aura{
-		Label:    "Impending Doom",
-		ActionID: core.ActionID{SpellID: 85107},
-		Duration: core.NeverExpires,
-		OnReset: func(aura *core.Aura, sim *core.Simulation) {
-			aura.Activate(sim)
-		},
-		//TODO: Do they need to hit?
-		OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
-			if (spell == warlock.ShadowBolt || spell == warlock.HandOfGuldan || spell == warlock.SoulFire || spell == warlock.Incinerate) &&
-				!warlock.Metamorphosis.CD.IsReady(sim) && sim.Proc(impendingDoomProcChance, "Impending Doom") {
-				*warlock.Metamorphosis.CD.Timer = core.Timer(time.Duration(*warlock.Metamorphosis.CD.Timer) - time.Second*15)
-				warlock.UpdateMajorCooldowns()
-			}
-		},
-	})
+	core.MakePermanent(
+		warlock.RegisterAura(core.Aura{
+			Label:    "Impending Doom",
+			ActionID: core.ActionID{SpellID: 85107},
+			//TODO: Do they need to hit?
+			OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+				if spell == warlock.ShadowBolt || spell == warlock.HandOfGuldan || spell == warlock.SoulFire || spell == warlock.Incinerate {
```

Comment:
You can use class masks for spell checks like those so you dont have to save references to spells in the warlock struct